### ISSUE TITLE
Make it clear the nopromises version in dist isn't "without promises API"

### ISSUE THREAD

**Issue by tofumatt** [mozilla/localForage#369]
As discussed in #368, it isn't clear that the `nopromises` files in the `dist/` directory are actually for users who are already including a Promises polyfill (or don't need one), not a version of localForage with the Promises API stripped out.

We should make this clear, in the README/wiki.

**Comment by zerotacg** [comment]
how about a "nopolyfill" version? that way the intention is clear and developers decide which polyfills to include, the requirements might need to be documented though

**Comment by thgreasi** [comment]
+1 to ".nopolyfills.js" version

On Fri, Jun 5, 2015 at 12:19 PM Tobias Peters <notifications@github.com>
wrote:

> how about a "nopolyfill" version? that way the intention is clear and
> developers decide which polyfills to include, the requirements might need
> to be documented though
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/localForage/issues/369#issuecomment-109214645>
> .
>
-- 
*Thodoris Greasidis*
Computer, Networking &
Communications Engineer

**Comment by SunboX** [comment]
Yep, this was confusing to me, too. Had to look inside the file to be sure that it isn't "without Promises"

**Comment by SunboX** [comment]
Btw, ".nopolyfills.js" may be confusing, too! Is ".nopolyfills.js" without `require()` polyfill?

Better: ".nopromises-polyfill.js"

**Comment by zerotacg** [comment]
```require()``` is polyfilled? looks like only promises are, require is handled by browserify, isn't it?

**Comment by mohitmayank** [comment]
+second that. Just wasted an hour trying to debug this. added minified no promises version to the build script.

**Comment by tofumatt** [comment]
I think in the future (2.0?) we should just scrap that version entirely. I don’t think the polyfill is that big of a size hit (14kB) and minified it’s only 5kB. Processing-wise it should be skipped if Promises are available, and it seems to only confuse people.


- tofumatt

**Comment by eric-brechemier** [comment]
I fell into that trap too. Our web application was not working on iPhone 4. It was due to the lack of the Promise API, expected by localforage.nopromises.js.

We are not using the external Promise API of localForage, and we wished to use a trimmed down version of the library because our use case is very simple and the library is disproportionately large compared to our usage (it is actually larger than all the rest of our code).

**Comment by nolanlawson** [comment]
>  think in the future (2.0?) we should just scrap that version entirely. I don’t think the polyfill is that big of a size hit (14kB) and minified it’s only 5kB.

+1 to this, especially if we switch to `lie` or to `native-promise-only` then the size of the Promise polyfill will be really negligible (between 1KB and 2KB min+gz, i.e. nothing).

**Comment by arxpoetica** [comment]
I just wasted an hour trying to figure this out. +1 to clarifying the name.
